File: flutter_app/lib/views/introduction_view.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:growthdeck/constants/mille_sanders_icons.dart';
import 'package:growthdeck/services/http_service.dart';

class IntroductionView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.white,
      child: SingleChildScrollView(
        physics: ClampingScrollPhysics(),
        scrollDirection: Axis.vertical,
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.center,
          children: <Widget>[
            Image(
              image: AssetImage('assets/images/duhastdenwillen.png'),
            ),
            Padding(
              padding: EdgeInsets.symmetric(horizontal: 20),
              child: Column(
                children: <Widget>[
                  SizedBox(height: 20),
                  RichText(
                    textAlign: TextAlign.left,
                    text: TextSpan(
                      style: DefaultTextStyle.of(context).style,
                      children: <TextSpan>[
                        TextSpan(
                            text:
                                'Danke, dass du Teil unserer Community bist!\n\n'),
                        TextSpan(
                            text:
                                'Mille Sanders dreht sich rund um die Themen '),
                        TextSpan(
                          text: 'Organisation, Produktivität und Planung.\n\n',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        TextSpan(text: 'Es geht um '),
                        TextSpan(
                          text: 'deine Ziele und Träume.',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        TextSpan(
                            text:
                                'Wir wollen dich motivieren diese zu erreichen und '),
                        TextSpan(
                          text:
                              'dir hilfreiche Tools dafür zur Verfügung stellen.\n\n',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        TextSpan(text: 'In unserem Sitz '),
                        TextSpan(
                          text: 'in Tirol ',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        TextSpan(
                            text:
                                'designen wir einige unserer Produkte und verwalten von dort aus auch unseren Shop.'),
                      ],
                    ),
                  ),
                  SizedBox(height: 20),
                  Center(
                    child: Column(
                      mainAxisSize: MainAxisSize.min,
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: <Widget>[
                        _BulletPoint(
                          icon: MilleSanders.noun_choose,
                          text: 'handverlesenes Sortiment',
                        ),
                        SizedBox(height: 10),
                        _BulletPoint(
                          icon: MilleSanders.noun_talk,
                          text: 'persöhnliche Beratung',
                        ),
                        SizedBox(height: 10),
                        _BulletPoint(
                          icon: MilleSanders.noun_shipping,
                          text: 'Versand nach DE und AT',
                        ),
                        SizedBox(height: 10),
                        _BulletPoint(
                          icon: MilleSanders.noun_environment,
                          text: 'klein & nachhaltig',
                        ),
                      ],
                    ),
                  ),
                  SizedBox(height: 20),
                  Center(
                    child: Row(
                      mainAxisSize: MainAxisSize.min,
                      children: <Widget>[
                        Text('Kontakt'),
                        SizedBox(width: 10),
                        GestureDetector(
                          onTap: () => HttpService.launchURL(
                              'http://m.me/millesandersaustria'),
                          child: Icon(
                            MilleSanders.iconfinder_facebook_messenger,
                            color: Colors.black,
                          ),
                        ),
                        SizedBox(width: 10),
                        GestureDetector(
                          onTap: () => HttpService.launchURL(
                              'mailto:contact@millesanders.com'),
                          child: Icon(
                            MilleSanders.iconfinder_mail,
                            color: Colors.black,
                          ),
                        ),
                      ],
                    ),
                  ),
                  SizedBox(height: 30),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}

class _BulletPoint extends StatelessWidget {
  _BulletPoint({this.icon, this.text});
  final IconData icon;
  final String text;

  @override
  Widget build(BuildContext context) => Row(
        mainAxisSize: MainAxisSize.min,
        mainAxisAlignment: MainAxisAlignment.start,
        children: <Widget>[
          Icon(
            icon,
            color: Colors.black,
            size: 30,
          ),
          SizedBox(width: 15),
          Text(text),
        ],
      );
}

File: ios_app/ios_app/IntroductionView.swift
//
//  IntroductionView.swift
//  ios_app
//
//  Created by Marco  Papula on 04.05.20.
//  Copyright © 2020 Marco  Papula. All rights reserved.
//

import SwiftUI

struct IntroductionView: View {
    var body: some View {
        ScrollView{
            Image("DuHastDenWillen")
                .resizable()
                .aspectRatio(contentMode: .fit)
                .padding(.bottom, 10.0)
            
            VStack(alignment: .leading) {
                Text("Danke, dass du Teil unserer Community bist!\n")

                Text("Mille Sanders dreht sich rund um die Themen ") +
                    Text("Organisation, Produktivität und Planung.\n").bold()
                
                Text("Es geht um deine ") +
                Text("Träume und Ziele. ").bold() +
                Text("Wir wollen dich motivieren, diese zu erreichen und ") +
                    Text("dir hilfreiche Tools dafür zur Verfügung stellen.\n").bold()
                
                (Text("In unserem Sitz ") +
                Text("in Tirol ") +
                Text("designen wir einige unserer Produkte und verwalten von dort aus auch unseren Shop."))
                
                VStack(alignment: .leading) {
                    IconBulletPoint(icon: "hand.raised", text:  "handverlesenes Sortiment")
                        
                    IconBulletPoint(icon: "person.2", text:  "persöhnliche Beratung")
                    
                    IconBulletPoint(icon: "paperplane", text:  "Versand nach DE und AT")
                    
                    IconBulletPoint(icon: "leaf.arrow.circlepath", text:  "klein & nachhaltig")
                    
                }
                .padding(.horizontal, 40.0)
                .padding(.bottom, 15)
                .padding(.top, 10)
                
            }.padding(.horizontal, 20.0)
                .font(.system(size: 15))
                .fixedSize(horizontal: false, vertical: true)
            
            HStack{
                Text("Kontakt")
                    .font(.system(size: 18))

                Image(systemName: "message.fill")
                    .resizable()
                    .scaledToFit()
                    .frame(width: 20, height: 20)
                    .onTapGesture {
                        if let url = URL(string: "http://m.me/millesandersaustria") {
                            UIApplication.shared.open(url)
                        }
                    }
                
                Image(systemName: "envelope.fill")
                    .resizable()
                    .scaledToFit()
                    .frame(width: 20, height: 20)
                    .onTapGesture {
                        if let url = URL(string: "mailto:contact@millesanders.com") {
                            UIApplication.shared.open(url)
                        }
                    }
            }
            .padding(.bottom, 20)
                        
        }
        .frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .top)
    }
}

struct IntroductionView_Previews: PreviewProvider {
    static var previews: some View {
        IntroductionView()
    }
}

struct IconBulletPoint: View{
    let icon: String;
    let text: String;
    
    var body: some View{
        HStack{
            Image(systemName: icon)
                .resizable()
                .scaledToFit()
                .frame(width: 20, height: 20)
                .padding(.trailing, 10)
            
            Text(text)
                .font(.custom("", size: 17))
            
        }
    }
}

